samples/net/http_server/src/main.c

```diff
@@ -0,0 +1,678 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
```

Comment:
The base of the application (minus the http protocol handling) seems to be taken from upstream
`samples/net/sockets/dumb_http_server_mt` sample. In this respect changing the license does not look right.
Perhaps it would make sense to send this application to upstream instead, there does not seem to be any Nordic specific stuff here.
cc: @carlescufi

Reply:
I forgot to include the orignal intel license, will add that. We are using the connection manager + net l2 backend for wifi which is only downstream. There might even be support for LTE at some point. But wifi was prioritized first.

Reply:
I don't how to properly handle this licensing. Some pointers here would do good. Maybe @ioannisg also have some ideas.

Reply:
There are two things here, copyright and license.
We should add original copyright together with the Nordic one, I do not see any issue here.
The more problematic one is the license which does not look to be compatible with Apache-2 so adding two licenses here feels questionable. But I am not a lawyer so this might be wrong.

About the sample itself. There is actually effort in upstream zephyr to create a HTTP server library and sample application (in PR 64465). I took the abandoned GSoC 2023 HTTP server project and continued its work on my free time. Unfortunately my progress have been very slow. Should I have known about this PR, I would have suggested that our efforts would have been concentrated enhancing the upstream version as it also supports HTTP/2.

Reply:
A HTTP server library would do good, the HTTP handling is quite raw. Avoiding direct socket and protocol handling would provide a nice abstraction. This will also make it easier for customers to add their own logic.

This sample was made in parallel with the wifi dev academy course, as some exercises in the course have been based on this sample. This gave us a good opportunity to make a _proper_ sample that could live in NCS,  supporting both wifi and lte. There has also been some interest in the wifi space for this sample in relation to home automation which has pushed the sense of urgency a bit.

But I guess the biggest blocker is the fact that an upstream version of this sample wouldn't support LTE, AFAIK.

Reply:
Regarding the license, I agree @carlescufi would be best to assist.

Regarding the sample itself:
- The HTTP server library would be very nice @jukkar. Without being an expert, I believe that any HTTP server sample (upstream or downstream) could be made to use this HTTP server library, once this library is available. 
- I do not see, however, much harm of having a dedicated downstream HTTP server sample (in addition to an upstream one). My thinking:
  - @simensrostad intended to use the connection manager library in the sample, and then share the sample code seamlessly between Wi-Fi and LTE. However, to the best of my knowledge, using the connectivity manager in upstream samples is of little value as there is no transport backend supported directly.
  - As mentioned, the upstream HTTP server example would not be directly usable for Nordic users/customers upstream (there is neither Nordic Wi-Fi, nor LTE support in the upstream project, for now)
  -  We could potentially scope or add functionality to this example that would be specific to Nordic needs, so it helps if the example is hosted in Nordic SDK.

Alternatively, one option would be to host the sample upstream, and support it on nRF70 Series in Nordic's downstream fork of Zephyr (with a Wi-Fi overlay, as @rlubos has done for other upstream Zephyr net examples..). But I am not sure how it could be made to work for LTE and nRF91 Series. Note that the original reason for creating the `samples/net` directory in `sdk-nrf` repository was to offer a set of NRF SDK networking examples, with a common code base for LTE and Wi-Fi using the connectivity (abstraction) manager feature.